### ISSUE TITLE
is.na warnings for version 5.4

### ISSUE THREAD

**Issue by liqg** [ropensci/drake#508]
When I updated the drake to drake_5.4.0.9001, I saw a lot of warnings about is.na(x), but it is OK for a previous version (v5.2).

```
> library(drake)
> plan <- drake_plan(
+   x = 1
+ )
> make(plan)
target x
Warning messages:
1: In is.na(x) : is.na() applied to non-(list or vector) of type 'NULL'
2: In is.na(x) : is.na() applied to non-(list or vector) of type 'NULL'
3: In is.na(x) : is.na() applied to non-(list or vector) of type 'NULL'
4: In is.na(x) : is.na() applied to non-(list or vector) of type 'NULL'
5: In is.na(x) : is.na() applied to non-(list or vector) of type 'NULL'
> sessionInfo()
R version 3.3.2 (2016-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 14.04.4 LTS

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
 [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] bindrcpp_0.2.2   drake_5.4.0.9001

loaded via a namespace (and not attached):
 [1] igraph_1.2.2      Rcpp_0.12.18      bindr_0.1.1       magrittr_1.5     
 [5] tidyselect_0.2.4  R6_2.2.2          rlang_0.2.2       storr_1.2.0      
 [9] stringr_1.3.1     dplyr_0.7.6       globals_0.12.1    tools_3.3.2      
[13] parallel_3.3.2    R.oo_1.21.0       withr_2.1.2       assertthat_0.2.0 
[17] digest_0.6.16     tibble_1.4.2      crayon_1.3.4      purrr_0.2.5      
[21] formatR_1.5       R.utils_2.6.0     fs_1.2.6          codetools_0.2-14 
[25] glue_1.3.0        evaluate_0.11     stringi_1.2.4     pillar_1.1.0     
[29] R.methodsS3_1.7.1 future_1.9.0      listenv_0.7.0     pkgconfig_2.0.2  
```

**Comment by liqg** [comment]
Thanks for your quick reply!!
```
> library(drake)
> options(warn = 2) # treats warnings as errors so we can get a traceback
> library(drake)
> plan <- drake_plan(x = 1)
> make(plan)
Error in is.na(x) : 
  (converted from warning) is.na() applied to non-(list or vector) of type 'NULL'
> traceback()
21: doWithOneRestart(return(expr), restart)
20: withOneRestart(expr, restarts[[1L]])
19: withRestarts({
        .Internal(.signalCondition(simpleWarning(msg, call), msg, 
            call))
        .Internal(.dfltWarn(msg, call))
    }, muffleWarning = function() NULL)
18: .signalSimpleWarning("is.na() applied to non-(list or vector) of type 'NULL'", 
        quote(is.na(x)))
17: sort.int(x, na.last = na.last, decreasing = decreasing, ...)
16: sort.default(unique(deps))
15: sort(unique(deps))
14: eval(expr, envir, enclos)
13: eval(lhs, parent, parent)
12: sort(unique(deps)) %>% self_hash(config = config) %>% digest::digest(algo = config$long_hash_algo)
11: dependency_hash(target = target, config = config)
10: drake_meta(target = target, config = config)
9: build_check_store(target = targets[i], config = config, downstream = targets[-seq_len(i)], 
       flag_attempt = TRUE)
8: run_loop(config = config)
7: get(paste0("run_", config$parallelism), envir = getNamespace("drake"))(config)
6: run_parallel_backend(config = config)
5: make_imports_targets(config = config)
4: make_with_schedules(config = config)
3: make_session(config = config)
2: make_with_config(config = config)
1: make(plan)
```

**Comment by wlandau** [comment]
Thanks, that is useful. Would you try the [`i508` branch](https://github.com/ropensci/drake/tree/i508)? There is a chance it could remove those warnings. If they are still there, I will install R 3.3.2 so I can reproduce what you are seeing.

**Comment by wlandau** [comment]
Just realized I forgot to actually push the change. 369d0d80fb4a48b7bbbc2a24cccc18a93a6c103f is what I recommend you try.

**Comment by liqg** [comment]
After running this, I also got the bug.
```R
devtools::install_github("ropensci/drake", ref="i508")
```

```
> library
> options(warn = 2) # treats warnings as errors so we can get a traceback
> library(drake)
> plan <- drake_plan(x = 1)
> make(plan)
Error in is.na(x) : 
  (converted from warning) is.na() applied to non-(list or vector) of type 'NULL'
> traceback() # le
21: doWithOneRestart(return(expr), restart)
20: withOneRestart(expr, restarts[[1L]])
19: withRestarts({
        .Internal(.signalCondition(simpleWarning(msg, call), msg, 
            call))
        .Internal(.dfltWarn(msg, call))
    }, muffleWarning = function() NULL)
18: .signalSimpleWarning("is.na() applied to non-(list or vector) of type 'NULL'", 
        quote(is.na(x)))
17: sort.int(x, na.last = na.last, decreasing = decreasing, ...)
16: sort.default(unique(deps))
15: sort(unique(deps))
14: eval(expr, envir, enclos)
13: eval(lhs, parent, parent)
12: sort(unique(deps)) %>% self_hash(config = config) %>% digest::digest(algo = config$long_hash_algo)
11: dependency_hash(target = target, config = config)
10: drake_meta(target = target, config = config)
9: build_check_store(target = targets[i], config = config, downstream = targets[-seq_len(i)], 
       flag_attempt = TRUE)
8: run_loop(config = config)
7: get(paste0("run_", config$parallelism), envir = getNamespace("drake"))(config)
6: run_parallel_backend(config = config)
5: make_imports_targets(config = config)
4: make_with_schedules(config = config)
3: make_session(config = config)
2: make_with_config(config = config)
1: make(plan)
```

**Comment by wlandau** [comment]
So apparently, `sort(NULL)` throws warnings on R 3.3, and that was causing the issue. I am diagnosing some other issues for `drake` on R 3.3, and I will push a patch today.

**Comment by wlandau** [comment]
Fixed via 15dcbde5554085bea12a6ad8e1a2ea8ebcf8303a.
